### DISCUSSION TITLE
Game History/Objectives

### DISCUSSION THREAD

**Discussion by Zac8668** [spicylobstergames/astratomic#31] [Ideas]
Opening a discussion about how the game history and objectives could be done.

First draft:

Final objective will be to reach the center of the galaxy.
Why? Don’t know yet

The first main objective will be to escape the first planet, for this you will need to create a first, rudimentary, spaceship.

The spaceship will follow through your entire journey, a system to improve its fuel capacity, space, velocity, etc should exist.
Your spaceship should be your “home”, even though you will probably have some bases on some planets.

To ensure you have bases in a diversity of planets, we will probably need a system like this for the travels:

IDEA 1: Each trip will cost fuel and processing power, more close to the center will mean it’s more costly for your spaceship computer to make a travel plan to the desired location. Previous traveled paths will be way cheaper to travel.

There needs to be a variety of planet types, hostile ones, biodiverse-friendly ones, resource full ones, etc
We could maybe have some planets that are more fit to farming, mining, etc, kinda like biomes.
But we also need to have variety within planets, if not it seems like we just moved the biomes from a main world to different planets.

In Starbound there is asteroid fuel mining, where it’s dangerous but you need to go there to get fuel, we should probably have a planet/moon type for this.

**Comment by erlend-sh** [comment]
## Galaxy

Will the planets of the galaxy be predetermined? I'm leaning towards yes. While the planets themselves can leave a lot up to procedural generation (though based on seeds that determine what resources are available on each), how the planets are lined up will determine the overarching pace of progression in the game, and this isn't something you wanna leave up to chance.

p.s. Are you sure a solar system wouldn't suffice? 😁

**Reply by Zac8668** [reply]
I think not, the idea to have a galaxy is to have a virtually infinite space for exploration, instead of infinite worlds like some games do, we would have finite worlds, but a virtually infinite amount of them, but we can add some way to ensure we have what we need to the game progression close to us. We can also make some "types" of solar systems to be predetermined.

**Comment by erlend-sh** [comment]
## Rocketship

> The first main objective will be to escape the first planet, for this you will need to create a first, rudimentary, spaceship.
> 
> The spaceship will follow through your entire journey, a system to improve its fuel capacity, space, velocity, etc should exist.
Your spaceship should be your “home”, even though you will probably have some bases on some planets.

Maybe the first planet just requires *fixing* the spaceship that's already there? This would prevent the first planet from feeling like you've already peaked by having built your one and only spaceship already, with only maintenance/upgrades to look forward to from there. A more exciting progression could be:

1. Fix spaceship
2. Upgrade spaceship for a few cycles; grow attached to it
3. Spaceship is destroyed! You're stranded.
4. Build a new spaceship from scratch.

> IDEA 1: Each trip will cost fuel and processing power, more close to the center will mean it’s more costly for your spaceship computer to make a travel plan to the desired location. Previous traveled paths will be way cheaper to travel.

Add-on idea: Every new planet you travel to, you build a Teleporter Station. Thus you can travel freely between any planets you've already been on, but to reach a new planet you have to use the rocket ship.

For extra trippiness factor, instead of Star Trek style teleporters, these could be merely *mind* teleporters. On each planet you've left behind a dormant copy of yourself, which you can remote-control through a teleporter. This could have implications for how save/load mechanics works, and also the extent to which materials can be moved between planets.

**Reply by Zac8668** [reply]
Really like the progression ideia! We could also use it as an excuse for an history, for example:
You crashed on the planet and broke your spaceship! You also lost all your memories of why you're there, fix your spaceship while getting increased help and insight of a spaceship ai npc, kinda like Jarvis.

Also, teleporters are pretty nice! Way better than the other idea.

**Reply by erlend-sh** [reply]
I see two straightforward possibilities for the ‘get to the center’ objective:

1. You’re trying to investigate something.
2. You’re making your way *back* to something, e.g. your home planet.

(2) has more built-in emotional impact so there’s a lot to like there.

(1) however has a simple way to align the story with the core game loop: Strange new types of organics have been discovered in this galaxy. You’ve been sent to investigate whether this life form is innately innocuous and without agenda, and where it came from.

**Comment by erlend-sh** [comment]
## Planets & Biomes

> There needs to be a variety of planet types, hostile ones, biodiverse-friendly ones, resource full ones, etc
> We could maybe have some planets that are more fit to farming, mining, etc, kinda like biomes. But we also need to have variety within planets, if not it seems like we just moved the biomes from a main world to different planets.

To an extent, yes. But beware of scope creep here. There's only so much we can do to make each planet distinct. Making an ecosystem is hard enough. Making *several* is a massive undertaking, especially if they're meant to emulate distinctly different biomes that have supposedly developed in their own special ways over vast timespans.

The constraints needed to keep the workload in check here will greatly influence the underlying story, and vice versa.

Do all these planets have a root life form in common? Was life created on these planets by design, or by accident? Did life develop over billions of years like our Earth did, or is there some accelerant in play?

**Reply by Zac8668** [reply]
I agree with you, we still have to see what we can achieve when we start doing proc gen.
